packages/app-store/stripepayment/lib/PaymentService.ts

```diff
@@ -92,11 +159,72 @@ export class PaymentService implements IAbstractPaymentService {
           fee: paymentFee,
           refunded: false,
           success: false,
+          paymentOption: paymentOption || "ON_BOOKING",
+        },
+      });
+      if (!paymentData) {
+        throw new Error();
+      }
+      return paymentData;
+    } catch (error) {
+      console.error(error);
+      throw new Error("Payment could not be created");
+    }
+  }
+
+  async chargeCard(payment: Payment, bookingId: Booking["id"]): Promise<Payment> {
+    try {
+      const stripeAppKeys = await prisma?.app.findFirst({
+        select: {
+          keys: true,
+        },
+        where: {
+          slug: "stripe",
         },
       });
+
+      const setupIntent = payment.data?.setupIntent;
+
+      // Parse keys with zod
+      const { client_id, payment_fee_fixed, payment_fee_percentage } = stripeAppKeysSchema.parse(
+        stripeAppKeys?.keys
+      );
+
+      const paymentFee = Math.round(payment.amount * payment_fee_percentage + payment_fee_fixed);
+
+      const params = {
+        amount: payment.amount,
+        currency: payment.currency,
+        application_fee_amount: paymentFee,
+        customer: setupIntent.customer,
+        payment_method: setupIntent.payment_method,
+        off_session: true,
+        confirm: true,
+      };
+
+      const paymentIntent = await this.stripe.paymentIntents.create(params, {
+        stripeAccount: this.credentials.stripe_user_id,
+      });
+
+      console.log(typeof payment.data);
+
+      const paymentData = await prisma.payment.update({
+        where: {
+          id: payment.id,
+        },
+        data: {
+          success: true,
+          data: {
+            ...payment.data,
+            paymentIntent,
```

Comment:
The stored data shouldn't replace the stored setup intent data we have in the DB